src/Misc.ino

```diff
@@ -1581,41 +1581,173 @@ String parseTemplate(String &tmpString, byte lineSize)
+                                  break;
+                                case 'I':
+                                  value = val == inverted ? "OUT" : "IN";
+                                  break;
+                                case 'Z' :// return "0" or "1"
+                                  value = val == inverted ? "0" : "1";
+                                  break;
+                                case '1' :
+                                    value = toString(valFloat,1);
+                                    break;
+                                case '2' :
+                                    value = toString(valFloat,2);
+                                    break;
+                                case '3' :
+                                    value = toString(valFloat,3);
+                                    break;
+                                case 'D' ://Dx.y min 'x' digits zero filled & 'y' decimal fixed digits
+                                  int x;
+                                  int y;
+                                  x = 0;
+                                  y = 0;
+
+                                  switch (tempValueFormat.length())
+                                  {
+                                    case 2: //Dx
+                                      if (tempValueFormat[1]>='0' && tempValueFormat[1]<='9')
+                                      {
+                                        x = (int)tempValueFormat[1]-'0';
+                                      }
+                                      break;
+                                    case 3: //D.y
+                                      if (tempValueFormat[1]=='.' && tempValueFormat[2]>='0' && tempValueFormat[2]<='9')
+                                      {
+                                        y = (int)tempValueFormat[2]-'0';
+                                      }
+                                      break;
+                                    case 4: //Dx.y
+                                      if (tempValueFormat[1]>='0' && tempValueFormat[1]<='9' && tempValueFormat[2]=='.' && tempValueFormat[3]>='0' && tempValueFormat[3]<='9')
+                                      {
+                                        x = (int)tempValueFormat[1]-'0';
+                                        y = (int)tempValueFormat[3]-'0';
+                                      }
+                                      break;
+                                    case 1: //D
+                                    default: //any other combination x=0; y=0;
+                                      break;
+                                  }
+                                  value = toString(valFloat,y);
+                                  int indexDot;
+                                  indexDot = value.indexOf('.') > 0 ? value.indexOf('.') : value.length();
+                                  for (byte f = 0; f < x - indexDot; f++)
```

Comment:
`f < x - indexDot` does work like it is intended here, but I always have to look up the [C++ operator precedence](http://en.cppreference.com/w/cpp/language/operator_precedence) when I see code like this.
It is more clear when using `()`.